docker/dockerfiles/be/upload_coredump.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/bash
+
+coredump_log()
+{
+    echo "[`date`] [coredump] $@" >&2
+}
+
+COREDUMP_PATH=/opt/starrocks/be/storage/coredumps
+
+if [ ! -d ${COREDUMP_PATH} ]; then
+    mkdir -p ${COREDUMP_PATH}
+fi
+
+cd $COREDUMP_PATH
+
+while true; do
+
+  latestCoreFile=""
+
+  # Setup inotifywait loop to wait until core file has been completely written
+  while read -r path action file; do
+    coredump_log "${action} ${path}${file}"
+
+    if [[ "$file" = core.*  ]]; then
+      latestCoreFile=$file
+    fi
+  done < <(inotifywait -e close_write $COREDUMP_PATH)
+
+  if [[ $latestCoreFile == "" ]]; then
+    coredump_log "missiong core file name"
+
+    continue
+  fi
+
+  coredump_log "Core dump generated $latestCoreFile"
+  coredump_log "$(ls -lh ${latestCoreFile})"
+
+  fileSize=$(ls -lh "${latestCoreFile}" | awk '{print $5}')
+  fileSizeBytes=$(stat -c %s "${latestCoreFile}")
+
+
+  # Check if file size is greater than 1TB (1000 GB)
+  if [[ "$fileSize" =~ T$ ]]; then
+      coredump_log "File size (${fileSizeBytes} B) exceeds 1TB: ${fileSizeBytes} bytes, skip"
+      rm ${latestCoreFile}
+
+      continue
+  fi
+
+
+  DATESTR=`TZ=${TZ} date +"%m-%d-%y.%H-%M-%S.%Z"`
```

Comment:
looks like `TZ=${TZ}` is not necessary if the `TZ` env is already there in current shell env.